Building swagger-core fails on Windows

Given OpenJDK and Apache Maven are available.

1. Clone swagger-core project.
2. Run `mvn -N`
3. Run `mvn install`

Expected: Build suceeds.

Actual: Build fails.

Build fails when running unit tests for the swagger-core module.

```
testBooleanSchemaSerialization(io.swagger.v3.core.serialization.OpenAPI3_1SerializationTest)  Time elapsed: 0 sec  <<< FAILURE!
java.lang.AssertionError: expected [{
  "openapi" : "3.1.0",
  "components" : {
    "schemas" : {
      "test" : true
    }
  }
}] but found [{
  "openapi" : "3.1.0",
  "components" : {
    "schemas" : {
      "test" : true
    }
  }
}]
        at org.testng.Assert.fail(Assert.java:99)
        at org.testng.Assert.failNotEquals(Assert.java:1037)
        at org.testng.Assert.assertEqualsImpl(Assert.java:140)
        at org.testng.Assert.assertEquals(Assert.java:122)
        at org.testng.Assert.assertEquals(Assert.java:629)
        at org.testng.Assert.assertEquals(Assert.java:639)
        at io.swagger.v3.core.serialization.OpenAPI3_1SerializationTest.testBooleanSchemaSerialization(OpenAPI3_1SerializationTest.java:1377)
```

OpenAPI3_1SerializationTest.java:1377:

```java
assertEquals(Json31.pretty(openAPI), "{\n" +
        "  \"openapi\" : \"3.1.0\",\n" +
        "  \"components\" : {\n" +
        "    \"schemas\" : {\n" +
        "      \"test\" : true\n" +
        "    }\n" +
        "  }\n" +
        "}");
```

`Json31.pretty(String)` uses Jackson's `DefaultIndenter` which uses the machine's line separator (`System.getProperty("line.separator")`), i.e. `"\r\n"` under Windows OS.

Swagger Core 2.2.5.SNAPHOT
Java, OpenJDK Runtime Environment 18.9 (build 11+28)
Windows 10.0.19045 Build 19045

--- a/modules/swagger-core/src/test/java/io/swagger/v3/core/serialization/OpenAPI3_1SerializationTest.java
+++ b/modules/swagger-core/src/test/java/io/swagger/v3/core/serialization/OpenAPI3_1SerializationTest.java
@@ -1,5 +1,6 @@
 package io.swagger.v3.core.serialization;
 
+import com.fasterxml.jackson.core.util.DefaultIndenter;
 import io.swagger.v3.core.matchers.SerializationMatchers;
 import io.swagger.v3.core.util.Json;
 import io.swagger.v3.core.util.Json31;
@@ -1374,14 +1375,14 @@ public class OpenAPI3_1SerializationTest {
 
         System.out.println("--------- root ----------");
         Json31.prettyPrint(openAPI);
-        assertEquals(Json31.pretty(openAPI), "{\n" +
+        assertEquals(Json31.pretty(openAPI), withJacksonSystemLineSeparator("{\n" +
                 "  \"openapi\" : \"3.1.0\",\n" +
                 "  \"components\" : {\n" +
                 "    \"schemas\" : {\n" +
                 "      \"test\" : true\n" +
                 "    }\n" +
                 "  }\n" +
-                "}");
+                "}"));
         System.out.println("--------- schema ----------");
         Json31.prettyPrint(openAPI.getComponents().getSchemas().get("test"));
         assertEquals(Json31.pretty(openAPI.getComponents().getSchemas().get("test")), "true");
@@ -1396,14 +1397,14 @@ public class OpenAPI3_1SerializationTest {
         assertEquals(Yaml31.pretty(openAPI.getComponents().getSchemas().get("test")), "true\n");
         System.out.println("--------- root 3.0 ----------");
         Json.prettyPrint(openAPI);
-        assertEquals(Json.pretty(openAPI), "{\n" +
+        assertEquals(Json.pretty(openAPI), withJacksonSystemLineSeparator("{\n" +
                 "  \"openapi\" : \"3.1.0\",\n" +
                 "  \"components\" : {\n" +
                 "    \"schemas\" : {\n" +
                 "      \"test\" : { }\n" +
                 "    }\n" +
                 "  }\n" +
-                "}");
+                "}"));
         System.out.println("--------- schema 3.0 ----------");
         Json.prettyPrint(openAPI.getComponents().getSchemas().get("test"));
         assertEquals(Json.pretty(openAPI.getComponents().getSchemas().get("test")), "{ }");
@@ -1418,4 +1419,7 @@ public class OpenAPI3_1SerializationTest {
         assertEquals(Yaml.pretty(openAPI.getComponents().getSchemas().get("test")), "{}\n");
     }
 
+    private static String withJacksonSystemLineSeparator(String s) {
+        return s.replace("\n", DefaultIndenter.SYS_LF);
+    }
 }

